### ISSUE TITLE
Post: SSH and GPG Relay inside WSL2 with systemd boot

### ISSUE THREAD

**Issue by lucasfcnunes** [twobiers/twobiers.github.io#3]
Comments for the post: https://twobiers.github.io/posts/wsl_systemd_gpg_ssh_relay

**Comment by lucasfcnunes** [comment]
@twobiers

- Where did you get the `npiperelay.exe` with the `-a` flag (Assuan)? https://github.com/albertony/npiperelay/pull/35
- It's possible to drop the use of `npiperelay.exe //./pipe/winssh-pageant` in favor of `npiperelay.exe -ei -s //./pipe/openssh-ssh-agent`. I tested it and it works. https://github.com/benpye/wsl-ssh-pageant/issues/58#issuecomment-1380753254

**Comment by demonbane** [comment]
@lucasfcnunes `-a` was added in @NZSmartie's [fork](https://github.com/NZSmartie/npiperelay). This is the only version that supports it to the best of my knowledge.

 If you want to set things up from scratch this is a great summary, but if you want to just get it working you can check out [wsl-gpg-systemd](https://github.com/demonbane/wsl-gpg-systemd) (I'm the author). No socat required, just npiperelay and systemd.

**Comment by lucasfcnunes** [comment]
> @lucasfcnunes `-a` was added in @NZSmartie's [fork](https://github.com/NZSmartie/npiperelay). This is the only version that supports it to the best of my knowledge.
> 
> If you want to set things up from scratch this is a great summary, but if you want to just get it working you can check out [wsl-gpg-systemd](https://github.com/demonbane/wsl-gpg-systemd) (I'm the author). No socat required, just npiperelay and systemd.

Here https://github.com/twobiers/twobiers.github.io/blob/b8720e95580689f8db40436f7fa74f90308fa188/content/posts/2023-07-23_wsl_systemd_gpg_ssh_relay/index.md?plain=1#L10 @twobiers point to @jstarks's.

**Comment by twobiers** [comment]
Good catch, thanks for pointing out. I've updated the link accordingly.

@demonbane You have come up with a cool solution there. Unfortunately I'm not using Windows/WSL at the moment to try it out in practice, but it looks great.

> While there are tutorials available for using some combination of socat, [wsl-ssh-pageant](https://github.com/benpye/wsl-ssh-pageant)/[wsl2-ssh-pageant](https://github.com/BlackReloaded/wsl2-ssh-pageant), and [npiperelay](https://github.com/jstarks/npiperelay), __they are a hassle to set up and very prone to breakage in my experience.__

True. I've set this up multiple times in various ways and eventually it broke.

**Comment by demonbane** [comment]
The changes from @NZSmartie's version of npiperelay have been merged into the [fork](https://github.com/albertony/npiperelay) maintained by @albertony. I've tested it in [wsl-gpg-systemd](https://github.com/demonbane/wsl-gpg-systemd) and everything is working fine. It's nice to not be relying on unmaintained code anymore.
